### DISCUSSION TITLE
add status == 'done' for status being 'success' or 'error'

### DISCUSSION THREAD

**Discussion by caub** [TanStack/query#2801] [Ideas] [answered]
That status would be error or success, possibly also a isDone field similar to isSuccess / isError

**Comment by TkDodo** [comment, marked as answer]
I think It’s a simple condition that you can do in user-land:

`const isDone = isSuccess || isError`

but I’m not sure when you’d actually want that? In most cases, this is just the opposite of isLoading

**Reply by caub** [reply]
@TkDodo thanks, yea right in the end what I needed is
```js
if (!step && orderMutation.isIdle) {
  orderMutation.mutate(formValue);
}
```
to make sure I run this only once (from the component)

so yes, no need for my suggestion
